File: ios/Claudiator/Models/AppNotification.swift
struct AppNotification: Codable, Identifiable, Hashable {
    var id: String { notificationId }
    let notificationId: String  // maps from "id" in JSON
    let sessionId: String
    let deviceId: String
    let title: String
    let body: String
    let notificationType: String
    let payloadJson: String
    let createdAt: String
    let acknowledged: Bool

    enum CodingKeys: String, CodingKey {
        case notificationId = "id"
        case sessionId, deviceId, title, body
        case notificationType, payloadJson, createdAt, acknowledged
    }
}

File: android/app/src/main/java/com/claudiator/app/models/AppNotification.kt
package com.claudiator.app.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class AppNotification(
    @SerialName("id") val notificationId: String,
    @SerialName("session_id") val sessionId: String,
    @SerialName("device_id") val deviceId: String,
    val title: String,
    val body: String,
    @SerialName("notification_type") val notificationType: String,
    @SerialName("payload_json") val payloadJson: String? = null,
    @SerialName("created_at") val createdAt: String,
    val acknowledged: Boolean? = null,
)
